core/src/main/java/brooklyn/catalog/internal/CatalogInitialization.java

```diff
@@ -0,0 +1,380 @@
+        this.isStartingUp = isStartingUp;
+    }
+
+    public void setFailOnStartupErrors(boolean startupFailOnCatalogErrors) {
+        this.failOnStartupErrors = startupFailOnCatalogErrors;
+    }
+
+    public CatalogInitialization addPopulationCallback(Function<CatalogInitialization, Void> callback) {
+        callbacks.add(callback);
+        return this;
+    }
+    
+    public ManagementContext getManagementContext() {
+        return Preconditions.checkNotNull(managementContext, "management context has not been injected into "+this);
+    }
+
+    public boolean isInitialResetRequested() {
+        return reset;
+    }
+
+    public boolean hasRunOfficial() { return hasRunOfficial; }
+    public boolean hasRunIncludingBestEffort() { return hasRunOfficial || hasRunBestEffort; }
+
+    /** makes or updates the mgmt catalog, based on the settings in this class */
+    public void populateCatalog(boolean needsInitial, Collection<CatalogItem<?, ?>> optionalItemsForResettingCatalog) {
+        synchronized (populatingCatalogMutex) {
+            try {
+                isPopulating = true;
+                BasicBrooklynCatalog catalog = (BasicBrooklynCatalog) managementContext.getCatalog();
+                if (!catalog.getCatalog().isLoaded()) {
+                    catalog.load();
+                } else {
+                    if (needsInitial && (hasRunOfficial || hasRunBestEffort)) {
+                        // an indication that something caused it to load early; not severe, but unusual
+                        log.warn("Catalog initialization has not properly run but management context has a catalog; re-populating, possibly overwriting items installed during earlier access (it may have been an early web request)");
+                        catalog.reset(ImmutableList.<CatalogItem<?,?>>of());
+                    }
+                }
+                hasRunOfficial = true;
+
+                populateCatalogImpl(catalog, needsInitial, optionalItemsForResettingCatalog);
+            } finally {
+                hasRunOfficial = true;
+                isPopulating = false;
+            }
+        }
+    }
+
+    private void populateCatalogImpl(BasicBrooklynCatalog catalog, boolean needsInitial, Collection<CatalogItem<?, ?>> optionalItemsForResettingCatalog) {
+        applyCatalogLoadMode();
+        
+        if (optionalItemsForResettingCatalog!=null) {
+            catalog.reset(optionalItemsForResettingCatalog);
+        }
+        
+        if (needsInitial) {
+            populateInitial(catalog);
+        }
+        
+        populateAdditions(catalog);
+
+        populateViaCallbacks(catalog);
+    }
+
+    private enum PopulateMode { YAML, XML, AUTODETECT }
+    
+    protected void populateInitial(BasicBrooklynCatalog catalog) {
+        if (disallowLocal) {
+            if (!hasRunOfficial()) {
+                log.debug("CLI initial catalog not being read with disallow-local mode set.");
+            }
+            return;
```

Comment:
i see `initialUri` as often being embedded in a script, so that deliberately it only has effect on a new catalog-initializing run. so it is normal for options to be combined to mask it, hence to warning or failure. the log will say where catalog is being loaded from, and so it shouldn't be too hard to work out that the initial uri is not being applied, and the description is clear why it wouldn't be (it's not the initial run).  also `CATALOG_LOAD_MODE` is now deprecated; it had a short shelf life so unlikely this code path will get executed.